src/lighting.h

```diff
@@ -46,11 +46,33 @@ struct AutoExposure
 	AutoExposure();
 };
 
+/**
+ * Parameters for set color and intensity of night and day light.
+ *
+ * Light color is calculated in function get_sunlight_color.
+ * Variable daynight_ration can be from 0 to 1000.
+ *
+ * sunlight->r = colorOffset_rgb[0]+colorRatioCoef_rgb[0]*daynight_ratio;
+ * sunlight->g = colorOffset_rgb[1]+colorRatioCoef_rgb[1]*daynight_ratio;
+ * sunlight->b = colorOffset_rgb[2]+colorRatioCoef_rgb[2]*daynight_ratio;
+ *
+ */
+struct LightIntensity
+{
+	/// @brief Sunlight color offset
+	float colorOffset_rgb[3];
```

Comment:
`f32`? (also, why not use `v3f`?)

Reply:
Because v3f is X, Y, Z vector. 
The use of video::SColor will be an alternative, but it includes alpha canal, which I do not need here.

Reply:
X, Y, Z are just names, like `[0]`, `[1]` and `[2]` for an array.

My point is just that conceptually, this value is a vector, and you're doing vector operations on it later on. You could cut on the boilerplate and reuse vector methods if you simply treated it as a vector.

I don't really see a reason *not* to represent colors as vectors. We already represent all different kinds (positions, velocities, accelerations, scales, etc.) of vectors as `v3f`s, some of which are way more questionable (like Euler angles).

I don't think `SColor` an option because it stores R, G, B, A in bytes (rather than floats). This also isn't really a color: R, G, B aren't in [0, 1] or [0, 255] bounds?

Reply:
Optimization purposes sound like a good argument to exchange.
Type `video::SColorf` should store colors in floats. Isn't it better to be used than v3f?

Reply:
> Optimization purposes sound like a good argument to exchange. Type `video::SColorf` should store colors in floats. Isn't it better to be used than v3f?

I don't think so, because as said, the values here aren't valid colors (and they don't make use of alpha, as you pointed out).